src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/DivInstruction.cs

```diff
@@ -189,186 +189,4 @@ public static Instruction Create(Type type)
             }
         }
     }
-
-    internal abstract class ModuloInstruction : Instruction
-    {
-        private static Instruction s_int16, s_int32, s_int64, s_UInt16, s_UInt32, s_UInt64, s_single, s_double;
-
-        public override int ConsumedStack => 2;
-        public override int ProducedStack => 1;
-        public override string InstructionName => "Modulo";
-
-        private ModuloInstruction() { }
-
-        private sealed class ModuloInt32 : ModuloInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object l = frame.Data[frame.StackIndex - 2];
-                object r = frame.Data[frame.StackIndex - 1];
-                if (l == null || r == null)
-                {
-                    frame.Data[frame.StackIndex - 2] = null;
-                }
-                else
-                {
-                    frame.Data[frame.StackIndex - 2] = ScriptingRuntimeHelpers.Int32ToObject((int)l % (int)r);
-                }
-                frame.StackIndex--;
-                return 1;
-            }
-        }
-
-        private sealed class ModuloInt16 : ModuloInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object l = frame.Data[frame.StackIndex - 2];
-                object r = frame.Data[frame.StackIndex - 1];
-                if (l == null || r == null)
-                {
-                    frame.Data[frame.StackIndex - 2] = null;
-                }
-                else
-                {
-                    frame.Data[frame.StackIndex - 2] = (short)((short)l % (short)r);
-                }
-                frame.StackIndex--;
-                return 1;
-            }
-        }
-
-        private sealed class ModuloInt64 : ModuloInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object l = frame.Data[frame.StackIndex - 2];
-                object r = frame.Data[frame.StackIndex - 1];
-                if (l == null || r == null)
-                {
-                    frame.Data[frame.StackIndex - 2] = null;
-                }
-                else
-                {
-                    frame.Data[frame.StackIndex - 2] = (long)l % (long)r;
```

Comment:
Just curious: all of these instructions follow the same form... did we consider just having a virtual for this one part of the operation, e.g.
```C#
internal abstract class ModuloInstruction : Instruction
{
    public override int Run(InterpretedFrame frame)
    {
        object l = frame.Data[frame.StackIndex - 2];
        object r = frame.Data[frame.StackIndex - 1];
        if (l == null || r == null)
        {
            frame.Data[frame.StackIndex - 2] = null;
        }
        else
        {
            frame.Data[frame.StackIndex - 2] = Mod(l, r);
        }
        frame.StackIndex--;
        return 1;
    }

    protected abstract object Mod(object l, object r);

    private sealed class ModuloInt32 : ModuloInstruction
    {
        protected override object Mod(object l, object r) => ScriptingRuntimeHelpers.Int32ToObject((int)l % (int)r);
    }

    private sealed class ModuloUInt16 : ModuloInstruction
    {
        protected override object Mod(object l, object r) => (ushort)((ushort)l % (ushort)r);
    }

    ... // etc.
}
```
?  Does that not work for some reason?  Or maybe the extra virtual call is prohibitively expensive in the scenario?  Seems like it would save a ton of code.